Lib/argparse.py

```diff
@@ -2569,17 +2565,14 @@ def _check_value(self, action, value):
             }
             msg = _('invalid choice: %(value)r (choose from %(choices)s)')
 
-            if self.suggest_on_error:
-                try:
-                    import difflib
-                    closest_choice = difflib.get_close_matches(value, action.choices, 1)
-                    if closest_choice:
-                        closest_choice = closest_choice[0]
-                        args['closest'] = closest_choice
-                        msg = _('invalid choice: %(value)r, maybe you meant %(closest)r? '
-                                '(choose from %(choices)s)')
-                except TypeError:
-                    closest_choice = []
+            if self.suggest_on_error and isinstance(value, str):  
+                import difflib
+                suggestions = difflib.get_close_matches(value, action.choices, 1)
```

Comment:
What if `choices` contain non-strings? Would not `get_close_matches()` fail? Perhapse this was the reason of `except TypeError` (but setting `closest_choice = []` was not needed).

I suggest to check also that all items of `choices` are strings. And please add a test.